src/box/iproto.cc

```diff
@@ -875,6 +907,12 @@ struct iproto_connection
 	struct rlist in_connections;
 	/** Set if connection is being dropped. */
 	bool is_drop_pending;
+	/**
+	 * Generation is sequence number of dropping connection invocation.
+	 *
+	 * See also `struct iproto_drop_finished`.
+	 */
+	int drop_generation;
```

Comment:
Let's make it `unsigned` to make it safe against overflow.

Reply:
Done.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/iproto.cc b/src/box/iproto.cc
index cb3ee2c77..9ab4b0d4b 100644
--- a/src/box/iproto.cc
+++ b/src/box/iproto.cc
@@ -150,7 +150,7 @@ struct iproto_drop_finished {
 	 * Message from previous invocation may be delivired and account
 	 * iproto thread as finished dropping connection which is not true.
 	 */
-	int generation;
+	unsigned generation;
 };
 
 struct iproto_thread {
@@ -260,7 +260,7 @@ static size_t drop_pending_thread_count;
  *
  * See also `struct iproto_drop_finished`.
  */
-static int drop_generation;
+static unsigned drop_generation;
 
 /**
  * IPROTO listen URIs. Set by box.cfg.listen.
@@ -442,7 +442,7 @@ struct iproto_cfg_msg: public cbus_call_msg
 			 *
 			 * See also `struct iproto_drop_finished`.
 			 */
-			int generation;
+			unsigned generation;
 		} drop_connections;
 	};
 	struct iproto_thread *iproto_thread;
@@ -912,7 +912,7 @@ struct iproto_connection
 	 *
 	 * See also `struct iproto_drop_finished`.
 	 */
-	int drop_generation;
+	unsigned drop_generation;
 	/**
 	 * Messaged sent to TX to cancel all inprogress requests of the
 	 * connection.
@@ -1689,7 +1689,8 @@ tx_process_drop_finished(struct cmsg *m)
 
 /** Send message to TX thread to notify that connections drop is finished. */
 static void
-iproto_send_drop_finished(struct iproto_thread *iproto_thread, int generation)
+iproto_send_drop_finished(struct iproto_thread *iproto_thread,
+			  unsigned generation)
 {
 	static const struct cmsg_hop drop_finished_route[1] =
 					{{ tx_process_drop_finished, NULL }};
```

</details>